modules/services/window-managers/i3-sway/sway.nix

```diff
@@ -0,0 +1,311 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.sway;
+
+  commonOptions = import ./lib/options.nix {
+    inherit lib cfg pkgs;
+    moduleName = "sway";
+    capitalModuleName = "Sway";
+  };
+
+  configModule = types.submodule {
+    options = {
+      inherit (commonOptions) fonts window floating focus assigns workspaceLayout modifier keycodebindings colors bars startup gaps;
+
+      left = mkOption {
+        type = types.str;
+        default = "h";
+        description = "Home row direction key for moving left.";
+      };
+
+      down = mkOption {
+        type = types.str;
+        default = "j";
+        description = "Home row direction key for moving down.";
+      };
+
+      up = mkOption {
+        type = types.str;
+        default = "k";
+        description = "Home row direction key for moving up.";
+      };
+
+      right = mkOption {
+        type = types.str;
+        default = "l";
+        description = "Home row direction key for moving right.";
+      };
+
+      terminal = mkOption {
+        type = types.str;
+        default = "${pkgs.rxvt_unicode}/bin/urxvt";
+        description = "Default terminal to run.";
+        example = "alacritty";
+      };
+
+      menu = mkOption {
```

Comment:
Seems like these options could be added to i3 module as well?

Reply:
Does the default i3 config use this construction? Would be happy to move it anyway, I don't think it will break anyone's config

Reply:
We could set it to `exec ${pkgs.dmenu}/bin/dmenu_run` for i3? And I was also thinking that it might make sense to move `left`, `down`, `right`, `up` and `terminal` options there if we want to declare them.